### PR TITLE
Allow to use --devenv without tox config

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/tox-dev/tox/issues/1643

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [x] added/updated test(s)
- [x] updated/extended the documentation
- [x] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/) in message body
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

File: src/tox/config/__init__.py [hunk 1]
```diff
@@ -1056,7 +1061,7 @@ class ParseIni(object):
     def __init__(self, config, ini_path, ini_data):  # noqa
         config.toxinipath = ini_path
         using("tox.ini: {} (pid {})".format(config.toxinipath, os.getpid()))
-        config.toxinidir = config.toxinipath.dirpath()
+        config.toxinidir = config.toxinipath.dirpath() if ini_path.check(file=True) else ini_path
```
Comment (hroncok) [inline src/tox/config/__init__.py:1064]: Note that this feels a bit like a hack. I've tried to pass `None` and adapt the code to not do certain things if this is `None`, but it was too messy.

**Comment by hroncok** [comment]
Typo in the commit message...

**Comment by hroncok** [comment]
> Typo in the commit message...

Amended.

**Comment by gaborbernat** [comment]
I'll defer review and merge on this to my co-maintainer @asottile as he's the champion of this feature.

**Pull request merged**